nexus/db-queries/tests/output/reserve_internal_ip_pool.sql

```diff
@@ -0,0 +1,38 @@
+UPDATE
+  ip_pool
+SET
+  reservation_type = $1, time_modified = now()
+WHERE
+  id = $2
+  AND time_deleted IS NULL
+  AND reservation_type = $3
+  AND (
+      SELECT
+        CAST(
+          IF(
+            EXISTS(
+              SELECT
+                1
+              FROM
+                external_ip
+              WHERE
+                ip_pool_id = $4 AND external_ip.is_service AND time_deleted IS NULL
+              LIMIT
+                1
+            ),
+            1 / 0,
+            1
+          )
+            AS BOOL
+        )
+    )
+  AND CAST(
+      IF(
+        (SELECT count(1) FROM ip_pool WHERE time_deleted IS NULL AND reservation_type = $5 LIMIT 2)
```

Comment:
Just thinking out loud - I think in practice this extra guard is probably not necessary? The guard above that we can't change the type of a pool if there are any services using IPs from it will prevent deleting any in-use pool, and there will always be at least one in-use pool (otherwise we wouldn't have any services with external IPs at all), right? But this seems super cheap and maybe makes it very clear that we always have to have at least one, so I'm on board with keeping it anyway.